### PR TITLE
Add: Initial support for bFLT v4

### PR DESCRIPTION
**Pull request by leberus**
Once again, this times relocations are fixed:

```
$ file test2
test2: BFLT executable - version 4 ram
$ r2 -e io.cache=true ./test2
[0x00000044]> s 0x00000160
[0x00000160]> pd 10
            0x00000160      14100be5       str r1, [fp, -0x14]
            0x00000164      30309fe5       ldr r3, [0x0000019c]        ; [0x19c:4]=0xdeadbeef
            0x00000168      08300be5       str r3, [fp, -8]
            0x0000016c      2c009fe5       ldr r0, [0x000001a0]        ; [0x1a0:4]=0x4668 str.Hola
            0x00000170      0d0000eb       bl 0x1ac
            0x00000174      28309fe5       ldr r3, [0x000001a4]        ; [0x1a4:4]=0x4660 str.Adios
            0x00000178      003093e5       ldr r3, [r3]
            0x0000017c      0310a0e1       mov r1, r3
            0x00000180      20009fe5       ldr r0, [0x000001a8]        ; [0x1a8:4]=0x4670 str._s_n
            0x00000184      1a0000eb       bl 0x1f4
[0x00000160]>
```

More should be needed, but so far is ok I think.
I'll see if i manage do install uClinux.

Another thing is that this plugin is able to manage bFLT version 4, but I'll port to v2 too.

### CODE REVIEW

File: libr/bin/p/bin_bflt.c [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+	.name = "bflt",
+	.desc = "bFLT format r_bin plugin",
+	.license = "LGPL3",
+	.get_sdb = NULL,
+	.load = &load,
+	.load_bytes = &load_bytes,
+	.destroy = &destroy,
+	.check = &check,
+	.check_bytes = &check_bytes,
+	.baddr = NULL,
```
Comment (radare) [inline libr/bin/p/bin_bflt.c]: you can skip those `.xxx = NULL`.

File: libr/bin/p/bin_bflt.c [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+		obj->reloc_table = reloc_table;
+	}
+
+out:
+	return list;
+}
+
+static RBinInfo *info(RBinFile *arch) {
+	struct r_bin_bflt_obj *obj = (struct r_bin_bflt_obj*)arch->o->bin_obj;
+	RBinInfo *info = R_NEW0(RBinInfo);
```
Comment (radare) [inline libr/bin/p/bin_bflt.c]: missing space

File: libr/bin/format/bflt/bflt.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#ifndef BFLT_H
+#define BFLT_H
+
+#include <r_types.h>
+#include <r_util.h>
+#include <r_lib.h>
+#include <r_bin.h>
+
+/* Version 4 */
+#define FLAT_VERSION            0x00000004L
```
Comment (radare) [inline libr/bin/format/bflt/bflt.h]: indent with spaces or tabs, not both

File: libr/bin/format/bflt/bflt.c [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+#include <r_util.h>
+#include <r_types.h>
+
+#include "bflt.h"
+
+#define READ(x, i) r_read_be32 (x + i); i += 4;
+
+RBinAddr *r_bflt_get_entry(struct r_bin_bflt_obj *bin) {
+        RBinAddr *addr = R_NEW0 (RBinAddr);
+
```
Comment (radare) [inline libr/bin/format/bflt/bflt.c]: remove empty line and negate the addr check like this (save 2 lines)

```
if (addr) {
  addr->paddr = bin->hdr->entry;
}
return addr;
```

File: libr/bin/format/bflt/bflt.c [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	if (!bflt_init_hdr (obj)) {
+		return false;
+	}
+	return true;
+}
+
+struct r_bin_bflt_obj *r_bin_bflt_new_buf(struct r_buf_t *buf) {
+	struct r_bin_bflt_obj *bin = R_NEW0 (struct r_bin_bflt_obj);
+	if (bin) {
```
Comment (radare) [inline libr/bin/format/bflt/bflt.c]: if (bin && r_bin_bflt_init ()..

File: libr/bin/format/bflt/bflt.c [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+		if (r_bin_bflt_init (bin, buf)) {
+			return bin;
+		}
+	}
+
```
Comment (radare) [inline libr/bin/format/bflt/bflt.c]: remove empty line

**Comment by radare** [comment]
Can you add a test case for radare2-regressions ? this pr lgtm, just misses those little changes

**Comment by Maijin** [comment]
Any bins/tests available for https://github.com/radare/radare2-regressions ?

**Comment by leberus** [comment]
I've addressed the issues. I'll add a test for bflt in radare2-regressions (during today)

**Comment by Maijin** [comment]
If you do another push pls fix the README too

**Comment by radare** [comment]
its merged

**Comment by alvarofe** [comment]
Please send a test to r2r

**Comment by leberus** [comment]
Hi @alvarofe I've created a test [here](https://github.com/leberus/radare2-regressions/commit/e6daed4f294225e5a1aa780ce04664834715c4f3) but for some reason is failing (I've tried the command manually and it works). Am I missing something? It's the first time i commit there so maybe I missed to edit some file.

**Comment by Maijin** [comment]
What do you mean it's failing? Any pastebin of the error?

**Comment by alvarofe** [comment]
Without more info i can't say but checkout the diff to find out what is the problem.

**Comment by leberus** [comment]
Sorry, i was in a hurry. Here there is the [pastebin](http://pastebin.com/V9JjkzDx)

**Comment by alvarofe** [comment]
create a PR just with the binary so i can test myself, since i did some changes in the code to be sure i didn't break anything

**Comment by Maijin** [comment]
Works here, I'm pushing

**Comment by Maijin** [comment]
done.

**Comment by leberus** [comment]
@alvarofe you didn't break anything, i tested the changes ;) (thanks)
@Maijin ok, maybe i had a poltergeist here, thanks anyway

**Pull request merged**